### PR TITLE
Fix missing as

### LINKED ISSUE #259: gcc: error trying to exec 'cc1': execvp: No such file or directory
https://travis-ci.org/iphydf/repro259/jobs/381693316#L1434

Repro: https://github.com/iphydf/repro259

### PR DESCRIPTION
**Pull request by mrkkrp**
Close #259.

@iphydf Here is a solution to the remaining problems described in #259. See descriptions of the problems in the readme and the workaronds. I did not include the fixes in the code because I felt that we should not hardwrie these workarounds into `rules_haskell`, but @m may have a different opinion, let me know what you think too.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,6 +200,60 @@ published on Hackage, or part of Stackage snapshots, using Bazel.
...
+  ```
+
+  It means that your `gcc` cannot find `as` by itself. This happens only on
+  certain operating systems such as Ubuntu. We need to make `as` visible
+  manually in that case:
+
+  ```bzl
+  # Create a symlink to system executable 'as'
+  genrule(
+      name = "as",
```
[the remark this answers is not in the document]
Comment (iphydf) [reply README.md]: It fixes the problem. I have other problems now, like linking errors, but this seems to work. I'll provide repros for my issues once this is merged.

File: README.md [hunk 2]
```diff
@@ -200,6 +200,60 @@ published on Hackage, or part of Stackage snapshots, using Bazel.
...
+
+* If you see error messages complaining about missing `as` executable:
+
+  ```
+  cc: error trying to exec 'as': execvp: No such file or directory
+  `cc' failed in phase `Assembler'. (Exit code: 1)
+  ```
+
+  It means that your `gcc` cannot find `as` by itself. This happens only on
+  certain operating systems such as Ubuntu. We need to make `as` visible
```
Comment (mboes) [inline README.md]: We should be more precise. Recent ubuntu's work fine. If anything it's Fedora that's more of a problem. In any case, what she should say is, this happens on certain platforms where GCC was built without the `--with-as` and `--with-ld` configure flags.

File: README.md [hunk 3]
```diff
@@ -200,6 +200,60 @@ published on Hackage, or part of Stackage snapshots, using Bazel.
 
 [hazel]: https://github.com/formationai/hazel
 
+## Troubleshooting
```
Comment (mboes) [inline README.md]: We should merge the two troubleshooting sections. No need to have two separate ones. Use the same format as the troubleshooting section below this one.

**Comment by mrkkrp** [comment]
@m PTAL.

**Comment by mrkkrp** [comment]
@mboes PTAL.

File: README.md [hunk 4]
```diff
@@ -200,7 +200,63 @@ published on Hackage, or part of Stackage snapshots, using Bazel.
...
+```
+
+It means that your `gcc` cannot find `as` by itself. This happens only on
+certain operating systems which have `gcc` compiled without `--with-as` and
+`--with-ld` flags. We need to make `as` visible manually in that case:
+
+```bzl
+# Create a symlink to system executable 'as'
+genrule(
+    name = "as",
```
Comment (iphydf) [inline README.md]: I've named this "toolchain_as". I'd recommend putting warning-free code in the example.

File: README.md [hunk 5]
```diff
@@ -200,7 +200,63 @@ published on Hackage, or part of Stackage snapshots, using Bazel.
 
 [hazel]: https://github.com/formationai/hazel
 
-## Troubleshooting `rules_haskell` development
+## Troubleshooting
+
+### `as`: no such file or directory
+
+If you see error messages complaining about missing `as` executable:
```
Comment (iphydf) [inline README.md]: Or `ld', if you like, or anything really (I had to put `touch` in there as well for `language-c` to compile - I don't know why).

**Comment by mrkkrp** [comment]
@iphydf Thanks for your feedback, I've amended the commit.

**Pull request merged**